homeworks/src/main/java/hw40/dao/OrderService.java

```diff
@@ -0,0 +1,46 @@
+package hw40.dao;
+
+import hw40.entity.Order;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Objects;
+
+@Service
+@RequiredArgsConstructor
+public class OrderService {
+    private final OrderRepository orderRepository;
+
+    public Order getOrderById(final Long id) {
+        validateOrderId(id);
+        return orderRepository.findById(id).orElse(null);
+    }
+
+    public List<Order> getAllOrders() {
+        return orderRepository.findAll();
+    }
+
+    @Transactional
```

Comment:
The best practice is to put `@Transactional` on the class level and make all the methods transactional by default. If your method is read-only, add a hint for the compiler - put `@Transactional(readOnly = true)` specifically for that method.